ProfessorAssessor/View Controllers/SignUpViewController.m

```diff
@@ -20,29 +23,64 @@ - (void)viewDidLoad {
     [super viewDidLoad];
 
     [self disableSignUpFieldsAndButtons];
+
+    [self saveFacebookAccountInformation];
 }
 
 - (IBAction)signUp:(UIButton *)sender {
+    if (self.email != nil) {
+        User *newUser = [self createUserWithFacebookLogin];
+
+        [self signUpNewUser:newUser];
+    } else {
+        User *newUser = [self createUserWithUsernameAndPassword];
+
+        if ([self validCredentials]) {
+            [self signUpNewUser:newUser];
+        }
+    }
+}
+
+- (User *)createUserWithFacebookLogin {
+    User *newUser = (User *)[PFUser user];
+
+    newUser.username = self.email;
+    newUser.password = self.email;
+    newUser.firstName = self.firstName;
+    newUser.lastName = self.lastName;
+    newUser.email = self.email;
+    newUser.school = self.school;
+
+    return newUser;
+}
+
+- (User *)createUserWithUsernameAndPassword {
     User *newUser = (User *)[PFUser user];
 
     newUser.username = self.username.text;
     newUser.password = self.password.text;
     newUser.school = self.school;
 
-    if ([self validCredentials]) {
-        [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-            if (succeeded) {
-                UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
-                HomeViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"HomeViewController"];
+    return newUser;
+}
 
-                SceneDelegate *sceneDelegate = (SceneDelegate *)self.view.window.windowScene.delegate;
+- (void)signUpNewUser:(User *)newUser {
```

Comment:
I like this separation of concerns. What's interesting is that you did this to account for facebook login which shows that this separation made up front would have saved you work. This is a great concrete example of how this principle helps keep code flexible.